diff --git a/src/.objects b/src/.objects
index b04dd03..419e6da 100644
--- a/src/.objects
+++ b/src/.objects
@@ -149,7 +149,11 @@ SOURCE=\
 	dos.o \
 	elf.o \
 	hamil_rot.o \
-	chain.o \
+  bfgs.o dynmat.o instanton.o lbfgs.o sd.o cg.o dimer.o bbm.o \
+  fire.o lanczos.o neb.o qm.o \
+  pyamff_fortran/*.o ml_pyamff.o \
+  opt.o	\
+  chain.o \
 	dyna.o \
 	fileio.o \
 	vhdf5.o \
diff --git a/src/main.F b/src/main.F
index 5a1980b..fae77bb 100644
--- a/src/main.F
+++ b/src/main.F
@@ -1046,7 +1046,7 @@ GPACC USE wpbe
 ! init all chains (INCAR reader)
 !-----------------------------------------------------------------------
       LCHAIN = IMAGES > 0 .AND. .NOT.AFQMC_SET % ACTIVE
-      IF (LCHAIN) CALL chain_init( T_INFO, IO)
+      CALL chain_init( T_INFO, IO)
 !-----------------------------------------------------------------------
 !xml finish copying parameters from INCAR to xml file
 ! no INCAR reading from here
@@ -3896,7 +3896,7 @@ io_end
          CALL CHAIN_STRESS( TSIF )
       END IF
       CALL CHAIN_FORCE(T_INFO%NIONS,DYN%POSION,TOTEN,TIFOR, &
-           LATT_CUR%A,LATT_CUR%B,IO%IU6)
+           TSIF,LATT_CUR%A,LATT_CUR%B,IO%IU6)
 
       CALL PARALLEL_TEMPERING(NSTEP,T_INFO%NIONS,DYN%POSION,DYN%VEL,TOTEN,TIFOR,DYN%TEBEG,DYN%TEEND, &
            LATT_CUR%A,LATT_CUR%B,IO%IU6)
diff --git a/src/makefile b/src/makefile
index ca5adc0..f1ed458 100644
--- a/src/makefile
+++ b/src/makefile
@@ -16,7 +16,7 @@ OFLAG_4=
 OFLAG=$(OFLAG_2)
 OFLAG_IN=$(OFLAG)
 
-LIB=lib parser
+LIB=lib parser pyamff_fortran
 LLIB=-Llib -ldmy -Lparser -lparser
 
 SRCDIR=../../src
@@ -174,7 +174,7 @@ vaspml:
 	cp $@/arch/makefile.include.vasp $@/makefile.include
 	$(MAKE) -C $@ libvaspml
 
-dependencies: sources
+dependencies: sources libs
 	$(MAKE) depend
 
 depend: $(F90SRC)
